tree/treeplayer/src/TDFUtils.cxx

```diff
@@ -124,18 +125,22 @@ ColumnName2ColumnTypeName(const std::string &colName, TTree *tree, TCustomColumn
    if (ds && ds->HasColumn(colName))
       colType = ds->GetTypeName(colName);
 
+   auto ComposeTVecTypeName = [](const std::string& valueType) {
+      return "ROOT::Experimental::VecOps::TVec<" + valueType + ">";
+      };
    TBranch *branch = tree ? tree->GetBranch(colName.c_str()) : nullptr;
    if (branch) {
       // this must be a real TTree branch
       static const TClassRef tbranchelRef("TBranchElement");
       if (branch->InheritsFrom(tbranchelRef)) {
          // this branch is not a fundamental type, we can ask for the class name
          std::string classname(static_cast<TBranchElement *>(branch)->GetClassName());
-         if (classname.compare(0, 7, "vector<") == 0) {
-            // column type is "vector<ValueType>", we read it as "TVec<ValueType>"
-            // value type is the classname.size() - 8 chars after the 7th character in "vector<ValueType>"
-            auto valueType = classname.substr(7, classname.size() - 8);
-            colType = "ROOT::Experimental::VecOps::TVec<" + valueType + ">";
+         if (ROOT::ESTLType::kSTLvector == TClassEdit::IsSTLCont(classname)) {
```

Comment:
Nope.  It returns an int that is a ROOT::ESTLType that is multiplied by -1 is the STL instance as a non-standard allocator.

Reply:
thanks @pcanal -- I thought streaming of STL containers with non-standard allocators was not supported, is that correct?

Reply:
It should be supported (and either way, we do need to detect the case :) ).

Reply:
The reason it ought to be supported is simply that the I/O for STL container is done via the creation of CollectionProxy which themselves uses wrapper functions that instantiated by the ROOT dictionary of the STL collection and thus it does not matter to the I/O what allocator is being used.

Reply:
Hi @pcanal. I think IO of vectors with non-standatd containers is not supported: we need a fix probably, or an upgrade in the direction of dictionaryless IO.
The reson why this fail is in two lines mainly (not only):
- Dictionaryless case (TEmulatedCollectionProxy): https://github.com/root-project/root/blob/master/io/io/src/TEmulatedCollectionProxy.cxx#L209
- Dictionary case: https://github.com/root-project/root/blob/master/io/io/src/TStreamerInfoActions.cxx#L1799

Basically the issue is that vector<T> and vector<T, MyAlloc<T>> do not have the same layout on all platforms, e.g. linux - on osx it looks like yes. The present behaviour is a segfault when these lines are executed.

I see several ways in which this could be fixed. In the emulated proxy case, we could avoid these superoptimised calls for the custom allocator case and use in this the interpreter - slower, more memory but at least it stands and does not crash (and we perfom IO too). For the compiled case we may need to avoid the optimised codepath in case of a custom allocator.

Reply:
> Basically the issue is that vector and vector<T, MyAlloc> do not have the same layout on all platforms, e.g. linux - on osx it looks like yes. 

Note that the code does not really on 'across platform similarly' but rely on the fact that for a given platform, all instance of std::vector have the same layout ... which could indeed be not true for custom vs non-custom .... However more likely than not, the real problem is that the short cut is not using the right allocator when extending (call to resize) the vector.

Note also that the Emulated case is supposed to be used only when the process does not have interpreted nor compiled code for the collection and thus *by definition* the allocator is irrelevant (Since anyway likely there is no code loaded for the custom allocator).   However it is likely that we remove the warning/error when using an Emulated Proxy for a compiled std::vector and we need to re-add it if this compiled vector has a custom allocator.

Reply:
HI,
I am not sure I get the point about the Emulated Proxy. Isn't it broken for the very same reason?

Reply:
yes, it is 'broken' for the same reason but the 'fix' ought to be different.  At first for all collections, when an emulated proxy is used to process a 'compiled' instance, we issue a warning and/or error message.  We later found that for std::vector the emulated proxy was actually dong the right thing and thus suppress the message for *all* vector.  We now have realized that for vector with a custom allocator this is not the case and we need to reintroduce the error message for vector with custom allocator.

Reply:
I see.